pytential/symbolic/execution.py

```diff
@@ -656,8 +675,36 @@ def __init__(self, places, sym_op_expr):
     def get_discretization(self, where):
         return self.places.get_discretization(where)
 
-    def get_modeled_cost(self, queue, **args):
-        cost_model_mapper = CostModelMapper(self, queue, args)
+    def cost_per_stage(self, queue, calibration_params, **args):
```

Comment:
I consider this the "public" interface for using the cost model, and I think separating `cost_per_stage` and `cost_per_box` explicitly as two methods is clearer. 

Elsewhere, I consider them the implementation details of the cost model, and they uses an argument flag instead of two functions for code reuse.